### PR TITLE
[CI] Use GitHub Actions

### PR DESCRIPTION
**Pull request by ikesyo**
The CI jobs are now about 2x faster than on Travis. Also adding Xcode 10.3 to the build matrix.

### CODE REVIEW

File: .github/workflows/integration_tests.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    tags:
+      - '*'
+  pull_request:
+
+jobs:
+  integration_tests:
+    runs-on: macOS-10.14
+    strategy:
+      matrix:
+        xcode: [10.1]
```
Comment (ikesyo) [inline .github/workflows/integration_tests.yml]: I'll add 10.3 in a separate pull request.

**Comment by ikesyo** [comment]
I've merged #2886 so rebase onto it.

**Comment by tmspzz** [comment]
will these be executed in parallel? (test with 10.3 are failing) 

```
Test Suite 'XcodeSpec' failed at 2019-10-04 01:50:18.118.
	 Executed 18 tests, with 8 failures (0 unexpected) in 199.612 (199.613) seconds
```

**Comment by kenji21** [comment]
I was having this on my MacBook long time ago, there is a framework not compiling with 10.3, I don't remember which one

**Comment by ikesyo** [comment]
Executed in parallel: https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobs

> A workflow run is made up of one or more jobs. Jobs run in parallel by default.

**Comment by ikesyo** [comment]
This is now passing ✅

**Comment by tmspzz** [comment]
@ikesyo thanks a lot!

**Pull request merged**